lib/tpm_kdfa.c

```diff
@@ -90,12 +90,26 @@ TPM_RC tpm_kdfa(TPMI_ALG_HASH hashAlg,
         return TPM_RC_HASH;
     }
 
-    HMAC_CTX ctx;
-    HMAC_CTX_init(&ctx);
-    int rc = HMAC_Init_ex(&ctx, key->buffer, key->size, md, NULL);
+    HMAC_CTX *ctx;
+#if OPENSSL_VERSION_NUMBER < 0x1010000fL /* OpenSSL 1.1.0 */
```

Comment:
@williamcroberts thanks a lot for your feedback, I've re-spun the branch and hopefully now it looks OK for you.